addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3583,13 +3583,18 @@ export class OdooEditor extends EventTarget {
             // Move selection if next character is zero-width space
             if (nextCharacter === '\u200B') {
                 focusOffset += 1;
-                while (focusNode && (!focusNode.textContent[focusOffset] || !closestElement(focusNode).isContentEditable)) {
-                    focusNode = nextLeaf(focusNode);
+                let newFocusNode = focusNode;
+                while (newFocusNode && (!newFocusNode.textContent[focusOffset] || !closestElement(newFocusNode).isContentEditable)) {
+                    newFocusNode = nextLeaf(newFocusNode);
                     focusOffset = 0;
                 }
-                const startContainer = ev.shiftKey ? selection.anchorNode : focusNode;
+                if (!focusOffset && closestBlock(focusNode) !== closestBlock(newFocusNode)) {
+                    newFocusNode = focusNode; // Do not move selection to next block.
+                    focusOffset = nodeSize(focusNode);
+                }
+                const startContainer = ev.shiftKey ? selection.anchorNode : newFocusNode;
                 const startOffset = ev.shiftKey ? selection.anchorOffset : focusOffset;
-                setSelection(startContainer, startOffset, focusNode, focusOffset);
+                setSelection(startContainer, startOffset, newFocusNode, focusOffset);
```

Comment:
I spent quite a lot of time trying to understand this commit, writing a comment to discuss the behavior, looking at the tests etc for things to finally click correctly in my head after a while and having to delete the comment so I think the commit message is not clear enough (or I'm just dumb).

When I read it, I thought we didn't want the arrow to skip to the next container, and I disagreed. Then I finally got it thanks to the tests. We do want that arrow keys skip the zws at the end of the node so that the user ends up in the next block, BUT the skipping mechanism _itself_ should not _already_ skip to the next block before the browser has its say otherwise it will skip the first character of the next block, essentially skipping _two_ arrow keystrokes instead of just one.

Do you think you could use your masterful english to make the distinction more obvious in the commit message please ? ^^'

Reply:
Haha indeed you got it right. Sure I'll work out a better description :-)